AudioManagerUtilities/src/CAmSocketHandler.cpp

```diff
@@ -287,31 +287,19 @@ am_Error_e CAmSocketHandler::listenToSignals(const std::vector<uint8_t> & listSi
         return (E_NOT_POSSIBLE);
     }
 
-    int signalHandlerFd;
+    sh_poll_s sgPollData;
     if(mSignalFdHandle)
     {
-      sh_poll_s sgPollData;
       if(E_OK!=getFDPollData(mSignalFdHandle, sgPollData))
       {
-         removeFDPoll(mSignalFdHandle);
-         mSignalFdHandle = 0;
-      }
-      else 
-      {
-        int signalHandlerFd = signalfd(sgPollData.pollfdValue.fd, &sigset, 0);
-        if (signalHandlerFd == -1)
-        {
-            logError("Could not update signal fd!");
-            return (E_NOT_POSSIBLE);
-        }
-        return E_OK;
+          mSignalFdHandle = 0;
       }
     }
     
     if(0==mSignalFdHandle)
     {
       /* Create the signalfd */
-      signalHandlerFd = signalfd(-1, &sigset, 0);
+      int signalHandlerFd = signalfd(-1, &sigset, SFD_NONBLOCK);
```

Comment:
I can see that this is following the original code but I would have chosen a variable that is not as similar: signalHandlerFd and mSignalFdHandle ...  Totally different things but they look quite similar.  (Still, since they are ints, looks like a chance these can be used wrong without compiler error.)

Anyway, now the variable scope is very short (just a few lines) so I think you don't need such a long name anymore.  "fd" might be just as good.

Reply:
Even I didn't like the implementation as of now. I have an updated version which I will provide next for this.
I will change to similar style I implemented for eventfd. Instead of keeping the handle as a member the implementation
will remember the fd id. This is also much faster because the lookup before is not needed anymore.